isis/src/control/apps/findfeatures/FastGeom.cpp

```diff
@@ -98,43 +477,98 @@ ImageTransform *FastGeom::compute(MatchImage &query, MatchImage &train)  {
     // same size as the output image below.
     if ( tSizeFull.area()  < (m_maxarea * qSize.area()) ) {
       // std::cout << "Use direct full mapping of Train-to-Query...\n";
-      fastg.reset(new GenericTransform("FastGeomMap", t_to_q,
+      fastg.reset(new GenericTransform("FastGeomMap", mapper,
                                        tSizeFull));
     }
   }
   else if ( "crop" == m_geomtype ) {
 
     // Crop train image to only common area in query image using inverse
     // std::cout << "Compute train to query BB...\n";
-    RectArea qbbox = ImageTransform::boundingBox( t_to_q.inv(), qSize,
-                                                   tSize.size());
+    RectArea qbbox = ImageTransform::boundingBox( GenericTransform::computeInverse(mapper),
+                                                  qSize,
+                                                  tSize.size());
     // std::cout << "TrainBB: " << qbbox << "\n";
 
 
     // std::cout << "Compute query to train BB...\n";
-    RectArea tbbox = ImageTransform::boundingBox( t_to_q, qbbox,
+    RectArea tbbox = ImageTransform::boundingBox( mapper, qbbox,
                                                    qSize.size());
 
     // std::cout << "MapBB: " << tbbox << "\n";
-    fastg.reset(new GenericTransform("FastGeomCrop", t_to_q, tbbox));
+    fastg.reset(new GenericTransform("FastGeomCrop", mapper, tbbox));
   }
 
  // If a mapper has not been allocated, allocate mapping to query image size
  // (as cam2cam does).
   if ( fastg.isNull() ) {
     // std::cout << "FastGeom cam2map map created...\n";
-    fastg.reset(new GenericTransform("FastGeomCamera", t_to_q, qSize.size()));
+    fastg.reset(new GenericTransform("FastGeomCamera", mapper, qSize.size()));
   }
 
   return ( fastg.take() );
 }
```

Comment:
This is a really big function doing a lot of things and has a lot of inputs. It would be good to split out some of the algorithms so that they can be properly unit tested and we can ensure that we're properly checking and handling all of the parameterization.

Reply:
I made separate methods for the grid, radial and dump portions of the compute() function and made them public scope so they can be directly called.